app/Filament/Resources/UserResource/Actions/AssignTeamBulkAction.php

```diff
@@ -0,0 +1,102 @@
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace App\Filament\Resources\UserResource\Actions;
+
+use Exception;
+use App\Models\User;
+use AdvisingApp\Team\Models\Team;
+use Filament\Support\Enums\MaxWidth;
+use Filament\Forms\Components\Select;
+use Filament\Tables\Actions\BulkAction;
+use Filament\Notifications\Notification;
+use Illuminate\Database\Eloquent\Collection;
+
+class AssignTeamBulkAction extends BulkAction
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->icon('heroicon-o-user-group')
+            ->modalWidth(MaxWidth::Small)
+            ->modalDescription(
+                fn (Collection $records): string => (count($records) > 1) ? 'This bulk action will overwrite any prior team assignments for the selected user(s).' : 'This bulk action will overwrite any prior team assignments for the selected user'
+            )
+            ->fillForm(fn (Collection $records): array => [
+                'records' => $records,
+            ])
+            ->form([
+                Select::make('team')
+                    ->label('Team')
+                    ->options(Team::pluck('name', 'id'))
+                    ->searchable(),
+            ])
+            ->action(function (array $data, Collection $records) {
+                $success = 0;
+                $fail = 0;
+                $records->each(function (User $record) use ($data, &$success, &$fail) {
+                    try {
+                        if (! empty($data['team'])) {
+                            $record->assignTeam($data['team']);
+                            $success = $success + 1;
+                        }
+                    } catch (Exception $e) {
+                        report($e);
+                        $fail = $fail + 1;
+                    }
+                });
+
+                if ($fail > 0) {
+                    Notification::make()
+                        ->title('Assigned Team')
+                        ->body('Fail to added to the team: ' . $fail)
+                        ->success()
+                        ->send();
+                } else {
+                    Notification::make()
+                        ->title('Assigned Team')
+                        ->success()
+                        ->send();
```

Comment:
We might want to add to the body here how many users were added to the team

Reply:
Done